docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,422 @@
+(which you can then register to be used by other visualizations).
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with the 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require any data to be requested.
+
+[[development-custom-request-handler]]
+==== custom request handler
+You can define your custom request handler by providing a function with the following definition:
+`function (vis, appState, uiState, searchSource) { ... }`
+
+Function must return a promise, which should get resolved with new data that will be passed to responseHandler.
+
+It's up to function to decide when it wants to issue a new request or return previous data 
+(if none of the objects relevant to the request handler changed).
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+const myRequestHandler = (vis, appState, uiState, searchSource) => {
+  const shouldQuery = () => {
+     // check if any of the relevant objects changed
+     // if not we should just return the old data
+  };
+  
+  return new Promise((resolve, reject) => {
+     if (shouldQuery()) {
+        // get your data
+        // resolve(myNewData);
+     } else {
+       resolve(myOldData);
+     }
+  });
+};
+
+const MyNewVisType = (Private) => {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    requestHandler: myRequestHandler
+  });
+}
+
+VisTypesRegistryProvider.register(MyNewVisType);
+-----------
+
+[[development-visualization-response-handlers]]
+=== Visualization Response Handlers
+Response handler will receive the data from request handler and an instance of Vis object. 
+Its job is to convert the data to a format visualization can use. By default 'default' request handler is used 
+which produces a table representation of the data. The data object will then be passed to visualization.
+
+[[development-default-response-handler]]
+==== default response handler
+Default response handler will convert pure elasticsearch response to tabular format.
+
+[[development-none-response-handler]]
+==== none response handler
+None response handler is an identity function, which will return the same data it receives.
+
+[[development-custom-response-handler]]
+==== custom response handler
+You can define your custom response handler by providing a function with the following definition:
+'function (vis, response) { ... }'.
+
+Function should return the transformed data object that visualization can consume.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+const myResponseHandler = (vis, response) => {
```

Comment:
I'd rewrite the example as:

```
const myResponseHandler = (vis, data) => {
   const response = .... transform data ....;
   return response;
}
```

I know, kinda stupid, but  I feel more details here would confuse people ;)